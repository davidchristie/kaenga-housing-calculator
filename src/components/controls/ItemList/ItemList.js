import React, { Component, PropTypes } from 'react'
import { Button, ListGroup } from 'react-bootstrap'

import Item from './Item'

class ItemList extends Component {
  render () {
    const { defaultItemValues, fields, itemComponent } = this.props
    return (
      <div>
        <Button onClick={() => fields.push(defaultItemValues)}>
          Add
        </Button>
        <ListGroup>
          {fields.map((item, index) =>
            <Item
              index={index}
              item={item}
              itemComponent={itemComponent}
              key={index}
              remove={() => {
                console.log(index)
                fields.remove(index)
              }}
            />
          )}
        </ListGroup>
        <Button onClick={() => fields.removeAll()}>
          Clear
        </Button>
      </div>
    )
  }
}

ItemList.defaultProps = {
  defaultItemValues: {},
  fields: {
    map: () => null
  }
}

ItemList.propTypes = {
  defaultItemValues: PropTypes.object.isRequired,
  fields: PropTypes.shape({
    map: PropTypes.func.isRequired
  }).isRequired,
  itemComponent: PropTypes.func
}

export default ItemList
