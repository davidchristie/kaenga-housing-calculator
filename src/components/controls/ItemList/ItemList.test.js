import React from 'react'
import ReactTestUtils from 'react-dom/test-utils'

import ItemList from './ItemList'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(
    shallowRenderer.render(<ItemList />)
  ).toMatchSnapshot()
})
