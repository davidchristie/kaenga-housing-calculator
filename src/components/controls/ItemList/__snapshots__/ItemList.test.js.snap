exports[`test matches snapshot 1`] = `
<div>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}>
    Add
  </Button>
  <ListGroup
    bsClass="list-group" />
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}>
    Clear
  </Button>
</div>
`;
